flink-connector-oracle-cdc/src/test/java/com/ververica/cdc/connectors/oracle/utils/OracleCdcContainer.java

```diff
@@ -0,0 +1,132 @@
+import org.testcontainers.images.builder.ImageFromDockerfile;
+
+/**
+ * Docker container for Oracle. The difference between this class and {@link
+ * org.testcontainers.containers.OracleContainer} is that TC OracleContainer has problems when using
+ * base docker image "wnameless/oracle-xe-11g-r2" (can't find log output "DATABASE IS READY TO
+ * USE!").
+ */
+public class OracleCdcContainer extends JdbcDatabaseContainer<OracleCdcContainer> {
+
+    public static final String NAME = "oracle";
+
+    private static final ImageFromDockerfile ORACLE_IMAGE =
+            new ImageFromDockerfile("oracle-xe-11g-tmp")
+                    .withFileFromClasspath(".", "docker")
+                    .withFileFromClasspath(
+                            "assets/activate-archivelog.sh", "docker/assets/activate-archivelog.sh")
+                    .withFileFromClasspath(
+                            "assets/activate-archivelog.sql",
+                            "docker/assets/activate-archivelog.sql");
+
+    private static final int ORACLE_PORT = 1521;
+    private static final int APEX_HTTP_PORT = 8080;
+
+    private static final int DEFAULT_STARTUP_TIMEOUT_SECONDS = 240;
+    private static final int DEFAULT_CONNECT_TIMEOUT_SECONDS = 120;
+
+    private String username = "system";
+    private String password = "oracle";
+
+    public OracleCdcContainer() {
+        super(ORACLE_IMAGE);
+        preconfigure();
+    }
+
+    private void preconfigure() {
+        withStartupTimeoutSeconds(DEFAULT_STARTUP_TIMEOUT_SECONDS);
+        withConnectTimeoutSeconds(DEFAULT_CONNECT_TIMEOUT_SECONDS);
+        addExposedPorts(ORACLE_PORT, APEX_HTTP_PORT);
+    }
+
+    @Override
+    protected Integer getLivenessCheckPort() {
+        return getMappedPort(ORACLE_PORT);
+    }
+
+    @Override
+    public String getDriverClassName() {
+        return "oracle.jdbc.OracleDriver";
+    }
+
+    @Override
+    public String getJdbcUrl() {
+        return "jdbc:oracle:thin:"
+                + getUsername()
+                + "/"
+                + getPassword()
+                + "@"
+                + getHost()
+                + ":"
+                + getOraclePort()
+                + ":"
+                + getSid();
+    }
+
+    @Override
+    public String getUsername() {
+        return username;
+    }
+
+    @Override
+    public String getPassword() {
+        return password;
+    }
+
+    @Override
+    public OracleCdcContainer withUsername(String username) {
+        this.username = username;
+        return self();
+    }
+
+    @Override
+    public OracleCdcContainer withPassword(String password) {
+        this.password = password;
+        return self();
+    }
+
+    @Override
+    public OracleCdcContainer withUrlParam(String paramName, String paramValue) {
+        throw new UnsupportedOperationException("The OracleDb does not support this");
+    }
+
+    @SuppressWarnings("SameReturnValue")
```

Comment:
Do we need this ?

Reply:
I just copied the source code from `testcontainers:1.15.1`.